[JIT] Get rid of incorrect shape analysis formulae

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22810 [JIT] Get rid of torch.mean shape analysis**


These formulae were wrong, since they were trying to figure out the number of elements in a const int list by checking the number of inputs to its node, but that's not valid

Differential Revision: [D16226973](https://our.internmc.facebook.com/intern/diff/D16226973)

Commit:
[JIT] Get rid of torch.mean shape analysis

Commit:
Update on "[JIT] Get rid of torch.mean shape analysis"

[JIT] Get rid of torch.mean shape analysis

gh-metadata: pytorch pytorch 22810 gh/jamesr66a/26/head

zdevito reviewed (approved):
Context: we rely less and less on shape propagation, and these formulas have a bug in them. Rather than spend time fixing the bug, we are just removing them. If we end up needed shape prop again in the future we can restore it.

Commit:
Update on "[JIT] Get rid of incorrect shape analysis formulae"

[JIT] Get rid of torch.mean shape analysis

gh-metadata: pytorch pytorch 22810 gh/jamesr66a/26/head
